src/openforms/submissions/logic/actions.py

```diff
@@ -190,29 +194,26 @@ class VariableAction(ActionOperation):
     def from_action(cls, action: ActionDict) -> "VariableAction":
         return cls(variable=action["variable"], value=action["action"]["value"])
 
-    def eval(self, context: DataMapping) -> DataMapping:
+    def eval(
+        self,
+        context: DataMapping,
+        log: Callable[[JSONValue], None],
+    ) -> DataMapping:
         with log_errors(self.value, self.rule):
-            return {self.variable: jsonLogic(self.value, context)}
+            log({"value": (value := jsonLogic(self.value, context))})
+            return {self.variable: value}
 
     def get_action_log_data(
         self,
         component_map: Dict[str, ComponentMeta],
         all_variables: Dict[str, FormVariable],
         initial_data: dict,
+        log_data: dict[str, JSONValue],
     ) -> JSONObject:
-        # Check if it's a primitive value, which doesn't require introspection
-        if not isinstance(self.value, (dict, list)):
-            value = self.value
-        else:
-            action_logic_introspection = introspect_json_logic(
-                self.value, component_map, initial_data
-            )
-            value = action_logic_introspection.as_string()
-
         return {
             "key": self.variable,
             "type_display": LogicActionTypes.get_label(LogicActionTypes.variable),
-            "value": value,
+            "value": log_data["value"],
```

Comment:
When calling `get_action_log_data` from `get_targeted_components` there is this:
```python
log_data=action_log_data.get(str(i), {})
```
So if the `log_data={}` I think this will give a key error.

Reply:
Oh yeah... It's when `eval` fails and the error is caught by the `log_errors`